force-app/main/default/classes/CBCreateSubscriptionTest.cls

```diff
@@ -1,397 +1,473 @@
-         pageRef.getParameters().put('con_id',coid);
-         pageRef.getParameters().put('domain','cbtest1');
-         Test.setMock(HttpCalloutMock.class, new MockCheckoutResponse1Generator());
-         CBCreateSubscription cbn = new CBCreateSubscription();
-         cbn.selectedSites='cbtest1';
-         cbn.plans='Testplan1';
-         cbn.planPrice = '100'; 
-         cbn.sDate = system.today();                
-         cbn.checkout();
-         Integer callouts = Limits.getCallouts();
-         Test.stopTest();
-         system.assertequals(1,callouts);
-     }
+        addOnSelected=cbs.selList[0];
+		addOnSelected.addon='StairStepaddon';	        
+        cbs.addonId=addOnSelected.Id;
+        cbs.CBPriceBookEntryAddon();        
+        expectedPriceChange=150.00;        
+        System.assertEquals(expectedPriceChange,addOnSelected.price);
+        System.assertEquals('Stairstep',addOnSelected.model);        
+    }
     
-    @IsTest public static void testSubscriptionVolType() 
-     {
-         Account acc = [select Id,Name from Account where Name='SfTest1' LIMIT 1];
-         system.debug('acc id'+acc.Id);
-         chargebeeapps__CB_Sites__c sites = [select name from chargebeeapps__CB_Sites__c  where name='cbtest1'];
-         Contact con = [select Id,LastName from Contact where LastName='test1' LIMIT 1];
-         chargebeeapps__Contact_Associate_Site__c testCon =[select chargebeeapps__External_Id__c,chargebeeapps__Contact__c,chargebeeapps__CB_Site__c from chargebeeapps__Contact_Associate_Site__c where chargebeeapps__Contact__c=:con.Id];
-    	 system.debug('testCon'+testCon);
-         Opportunity opt = [select Id,Name,StageName,CloseDate from Opportunity where Name='test opp1' LIMIT 1];
-         string oppid = opt.Id;
-         string coid = con.Id;
-         Product2[] prd = [select Name,ExternalId from Product2];
-         system.debug('prd--'+prd);   
-         PricebookEntry[] selctedprod = [select Id,Product2.Name,Product2.ExternalId from PricebookEntry];
-         system.debug('selctedprod--'+selctedprod);
-         //chargebeeapps__CB_Pricing_Tiers__c[] cbtiers;
+    public static Map<String,List<chargebeeapps__CB_Pricing_Tiers__c>> getPricingTiers(String productType)
+    {
+        Map<String,List<chargebeeapps__CB_Pricing_Tiers__c>> ptlist=new Map<String,List<chargebeeapps__CB_Pricing_Tiers__c>>();
+        String defaultSiteId = TestdataUtility.getSite('cbtest1').Id;
+        List<String> pricingmodeltypes = new List<String>{'Tiered','Stairstep','Volume'};
+        for(String thepricingmodel:pricingmodeltypes)
+        {
+            String productname=thepricingmodel+productType;
+            PriceBookEntry pbEntry = [Select Id,Product2.ExternalId,product2.chargebeeapps__Recurring_charge_model__c,product2.chargebeeapps__Addon_Type__c,product2.chargebeeapps__Product_Type__c,PricebookEntry.chargebeeapps__Setup_Cost__c,PricebookEntry.UnitPrice,product2.chargebeeapps__Currency__c 
```

Comment:
reuse the ProductService.getPricingTiers query.

Reply:
Used existing Api to calculate the Expected data wherever applicable.